metrics.nim

```diff
@@ -266,7 +248,9 @@ proc collect*(registry: Registry): OrderedTable[Collector, Metrics] =
     withLock registry.lock:
       for collector in registry.collectors:
         var collectorCopy: Collector
-        deepCopy(collectorCopy, collector)
+        withLock collector.lock:
+          # Warning: data race risk here.
```

Comment:
not with a lock in place - it might be prudent to describe where data that is being copied is updated though, and why it needs a lock

Reply:
Comment removed.